pxemgr/pxemanager.go

```diff
@@ -187,77 +195,97 @@ func withSerialParam(serialHandler func(serial string, w http.ResponseWriter, r
-	mgr.router.Methods("GET").PathPrefix("/hostinfo-helper").HandlerFunc(mgr.infoPusher)
+	mgr.pxeRouter.Methods("GET").PathPrefix("/hostinfo-helper").HandlerFunc(mgr.infoPusher)
 
 	if mgr.useIgnition {
-		mgr.router.Methods("POST").PathPrefix("/final-ignition-config.json").HandlerFunc(mgr.configGenerator)
+		mgr.pxeRouter.Methods("POST").PathPrefix("/final-ignition-config.json").HandlerFunc(mgr.configGenerator)
 	} else {
-		mgr.router.Methods("POST").PathPrefix("/final-cloud-config.yaml").HandlerFunc(mgr.configGenerator)
+		mgr.pxeRouter.Methods("POST").PathPrefix("/final-cloud-config.yaml").HandlerFunc(mgr.configGenerator)
 	}
 
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/boot_complete").HandlerFunc(withSerialParam(mgr.bootComplete))
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/set_installed").HandlerFunc(withSerialParam(mgr.setInstalled))
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/set_metadata").HandlerFunc(withSerialParam(mgr.setMetadata))
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/mark_fresh").HandlerFunc(withSerialParam(mgr.markFresh))
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/set_provider_id").HandlerFunc(withSerialParam(mgr.setProviderId))
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/set_ipmi_addr").HandlerFunc(withSerialParam(mgr.setIPMIAddr))
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/set_cabinet").HandlerFunc(withSerialParam(mgr.setCabinet))
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/set_state").HandlerFunc(withSerialParam(mgr.setState))
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/set_etcd_cluster_token").HandlerFunc(withSerialParam(mgr.setEtcdClusterToken))
-	mgr.router.Methods("PUT").PathPrefix("/admin/host/{serial}/override").HandlerFunc(withSerialParam(mgr.override))
+	// boring stuff
+	mgr.pxeRouter.Methods("GET").PathPrefix("/images/{serial}").HandlerFunc(mgr.imagesHandler)
+
+	// serve assets for yochu like etcd, fleet, docker, kubectl and rkt
+	mgr.pxeRouter.PathPrefix("/yochu").Handler(http.StripPrefix("/yochu", http.FileServer(http.Dir(mgr.yochuPath))))
+
+	// add welcome handler for debugging
+	mgr.pxeRouter.Path("/").HandlerFunc(mgr.welcomeHandler)
 
-	mgr.router.Methods("GET").PathPrefix("/admin/mayu_config").HandlerFunc(mgr.getMayuConfig)
-	mgr.router.Methods("PUT").PathPrefix("/admin/mayu_config").HandlerFunc(mgr.setMayuConfig)
+	// serve static files like infopusher and mayuctl etc.
+	mgr.pxeRouter.PathPrefix("/").Handler(http.FileServer(http.Dir(mgr.staticHTMLPath)))
+
+	glogWrapper := logging.NewGlogWrapper(8)
+	loggedRouter := handlers.LoggingHandler(glogWrapper, mgr.pxeRouter)
+
+	glog.V(8).Infoln(fmt.Sprintf("starting iPXE server at %s:%d", mgr.bindAddress, mgr.pxePort))
+
+	err := http.ListenAndServe(fmt.Sprintf("%s:%d", mgr.bindAddress, mgr.pxePort), loggedRouter)
+	if err != nil {
+		errChan <- mayuerror.MaskAny(err)
+		return
+	}
+	errChan <- nil
+}
```

Comment:
`http.ListenAndServe` blocks anyway until it perceives any error. There is no need for a channel. I would simply return an error in case one happens and the caller can handle it. [main.go](https://github.com/giantswarm/mayu/blob/3de7792b5bd4159c52d5d2f59a14cf63f74fd886/main.go#L283) does it already.